src/codegen/function.rs

```diff
@@ -50,17 +52,17 @@ pub fn declaration(library: &library::Library, analysis: &analysis::functions::I
     format!("fn {}({}){}", analysis.name, param_str, return_str)
 }
 
-macro_rules! write_to_vec(
-    ($dst:expr, $($arg:tt)*) => (
-        $dst.push(fmt::format(format_args!($($arg)*)))
-    )
-);
+pub fn body(library: &library::Library, analysis: &analysis::functions::Info,
+    in_trait: bool) -> Vec<String> {
+    let mut builder = Builder::new()
+        .glib_name(&analysis.glib_name)
+        .from_glib(analysis.ret.to_return_value_body(&library, &analysis));
 
-pub fn body(analysis: &analysis::functions::Info) -> Vec<String> {
-    let mut v: Vec<String> = Vec::new();
-    //TODO: real generation
-    write_to_vec!(v, "unsafe {{");
-    write_to_vec!(v, "{}TODO: call ffi:{}()", tabs(1), analysis.glib_name);
-    write_to_vec!(v, "}}");
-    v
+    //TODO: change to map on parameters with pass Vec<String> to builder
+    for par in &analysis.parameters {
+        let s = par.to_parameter_body(library, in_trait);
+        builder = builder.parameter(s);

```

Comment:
I guess this is some of the motivation for non-consuming builders: you'd just do `builder.parameter(s);` here.